zetaclient/bitcoin/bitcoin_client_test.go

```diff
@@ -212,142 +210,508 @@ func TestCalcDepositorFee828440(t *testing.T) {
-		_, cctx := testutils.LoadBTCTxRawResultNCctx(t, chainID, nonce)
+		_, cctx := testutils.LoadBTCTxRawResultNCctx(chainID, nonce)
 		params := cctx.GetCurrentOutTxParam()
 
-		err := btcClient.checkTSSVoutCancelled(params, []btcjson.Vout{})
+		err := btcClient.checkTSSVoutCancelled(params, []btcjson.Vout{}, chain)
 		require.ErrorContains(t, err, "invalid number of vouts")
 
-		err = btcClient.checkTSSVoutCancelled(params, []btcjson.Vout{{}, {}, {}})
+		err = btcClient.checkTSSVoutCancelled(params, []btcjson.Vout{{}, {}, {}}, chain)
 		require.ErrorContains(t, err, "invalid number of vouts")
 	})
 	t.Run("should fail if vout 0 is not to the TSS address", func(t *testing.T) {
 		// remove change vout to simulate cancelled tx
-		rawResult, cctx := testutils.LoadBTCTxRawResultNCctx(t, chainID, nonce)
+		rawResult, cctx := testutils.LoadBTCTxRawResultNCctx(chainID, nonce)
 		rawResult.Vout[1] = rawResult.Vout[2]
 		rawResult.Vout = rawResult.Vout[:2]
 		params := cctx.GetCurrentOutTxParam()
 
 		// not TSS address, bc1qh297vdt8xq6df5xae9z8gzd4jsu9a392mp0dus
 		rawResult.Vout[0].ScriptPubKey.Hex = "0014ba8be635673034d4d0ddc9447409b594385ec4aa"
-		err := btcClient.checkTSSVoutCancelled(params, rawResult.Vout)
+		err := btcClient.checkTSSVoutCancelled(params, rawResult.Vout, chain)
 		require.ErrorContains(t, err, "not match TSS address")
 	})
 	t.Run("should fail if vout 0 not match nonce mark", func(t *testing.T) {
 		// remove change vout to simulate cancelled tx
-		rawResult, cctx := testutils.LoadBTCTxRawResultNCctx(t, chainID, nonce)
+		rawResult, cctx := testutils.LoadBTCTxRawResultNCctx(chainID, nonce)
 		rawResult.Vout[1] = rawResult.Vout[2]
 		rawResult.Vout = rawResult.Vout[:2]
 		params := cctx.GetCurrentOutTxParam()
 
 		// not match nonce mark
 		rawResult.Vout[0].Value = 0.00000147
-		err := btcClient.checkTSSVoutCancelled(params, rawResult.Vout)
+		err := btcClient.checkTSSVoutCancelled(params, rawResult.Vout, chain)
 		require.ErrorContains(t, err, "not match nonce-mark amount")
 	})
 	t.Run("should fail if vout 1 is not to the TSS address", func(t *testing.T) {
 		// remove change vout to simulate cancelled tx
-		rawResult, cctx := testutils.LoadBTCTxRawResultNCctx(t, chainID, nonce)
+		rawResult, cctx := testutils.LoadBTCTxRawResultNCctx(chainID, nonce)
 		rawResult.Vout[1] = rawResult.Vout[2]
 		rawResult.Vout = rawResult.Vout[:2]
 		params := cctx.GetCurrentOutTxParam()
 
 		// not TSS address, bc1qh297vdt8xq6df5xae9z8gzd4jsu9a392mp0dus
 		rawResult.Vout[1].ScriptPubKey.Hex = "0014ba8be635673034d4d0ddc9447409b594385ec4aa"
-		err := btcClient.checkTSSVoutCancelled(params, rawResult.Vout)
+		err := btcClient.checkTSSVoutCancelled(params, rawResult.Vout, chain)
 		require.ErrorContains(t, err, "not match TSS address")
 	})
 }
 
+// createRPCClientAndLoadTx is a helper function to load raw tx and feed it to mock rpc client
+func createRPCClientAndLoadTx(chainId int64, txHash string) *stub.MockBTCRPCClient {
```

Comment:
I think we can put all utility functions at the top of the test file for better organization

Reply:
https://github.com/zeta-chain/node/pull/1942/commits/2bcdfbc727c12612159ffbf008bb00505a996b4b